File: src/main/java/no/nav/sosialhjelp/soknad/oppslag/KontonummerService.java
package no.nav.sosialhjelp.soknad.oppslag;

import org.springframework.stereotype.Component;

@Component
public class KontonummerService {

    private final OppslagConsumer kontonummerConsumer;

    public KontonummerService(OppslagConsumer kontonummerConsumer) {
        this.kontonummerConsumer = kontonummerConsumer;
    }

    public String getKontonummer(String ident) {
        var kontonummer = kontonummerConsumer.getKontonummer(ident);
        if (kontonummer == null) {
            return null;
        }
        return kontonummer.getKontonummer();
    }
}

File: src/main/kotlin/no/nav/sosialhjelp/soknad/kontonummer/KontonummerService.kt
package no.nav.sosialhjelp.soknad.kontonummer

open class KontonummerService(
    private val kontonummerClient: KontonummerClient
) {

    open fun getKontonummer(ident: String): String? {
        return kontonummerClient.getKontonummer(ident)?.kontonummer
    }
}
